ZelBack/src/services/fluxNetworkHelper.js

```diff
@@ -170,91 +168,18 @@ function isPortBanned(port) {
 
 /**
  * To perform a basic check if port on an ip is opened
- * This requires our port to be also open on out
  * @param {string} ip IP address.
  * @param {number} port Port.
- * @param {string} app Application name. Mostly for comsetic purposes, can be boolean. Defaults to undefined, as for testing main FluxOS not an app.
- * @param {number} timeout Timeout in ms.
  * @returns {boolean} Returns true if opened, otherwise false
  */
-async function isPortOpen(ip, port, app, timeout = 5000) {
-  let resp;
-  let firewallActive;
+async function isPortOpen(ip, port) {
   try {
-    let portResponse = true;
-    // eslint-disable-next-line no-use-before-define
-    firewallActive = await isFirewallActive();
-    // open port first
-    if (firewallActive) {
-      // eslint-disable-next-line no-use-before-define
-      resp = await allowOutPort(port).catch((error) => { // requires allow out for apps checking, for our ports both
-        log.error(error);
-      });
-      if (!resp) {
-        resp = {};
-      }
-    } else {
-      resp.message = 'existing';
-    }
-
-    const promise = new Promise(((resolve, reject) => {
-      const socket = new net.Socket();
-
-      const onError = (err) => {
-        socket.destroy();
-        if (err.code && err.code === 'ETIMEDOUT') {
-          log.info(`Connection on ${ip}:${port} ETIMEDOUT. Flux or Flux App is not running correctly`);
-          reject();
-        } else if (app) {
-          resolve();
-        } else if (port === 16129) {
-          log.error(`Syncthing of Flux on ${ip}:${port} did not respond correctly but may be in use. Allowing`);
-          log.error(err);
-          resolve();
-        } else {
-          log.error(`Flux on ${ip}:${port} is not working correctly`);
-          log.error(err);
-          reject();
-        }
-      };
-
-      const onTimeout = () => {
-        log.error(`Connection on ${ip}:${port} timed out. Flux or Flux App is not running correctly`);
-        socket.destroy();
-        reject();
-      };
-
-      socket.setTimeout(timeout);
-      socket.once('error', onError);
-      socket.once('timeout', onTimeout);
-
-      socket.connect(port, ip, () => {
-        socket.destroy();
-        portResponse = 'listening';
-        resolve();
-      });
-    }));
-    await promise;
-    setTimeout(() => { // timeout ensure return first
-      if (app && firewallActive) {
-        // delete the rule
-        if (resp.message !== 'existing') { // new or updated rule or firewall not active from above
-          // eslint-disable-next-line no-use-before-define
-          deleteAllowOutPortRule(port); // no need waiting for response. Delete if was not present before to not create huge firewall list
-        }
-      }
-    }, 10);
-    return portResponse; // true for OK port. listening for port that is being listened to
+    const exec = `nc -w 5 -z -v ${ip} ${port} </dev/null; echo $?`;
```

Comment:
Tested many scenarios. It does not work well
This function should only verify if a port is open or not and this modification does not work
My test IP was 89.58.42.150
Works well on its default ports 16126, 16127, 16128, 16129. But fails on other ports that are open too such as 16137.
More importantly it fails on ports that are used by applications such as presearch. There is presearch running on port 39000 and it gives connection refused as well.
Perhaps failiure should only be considered if it returns timed out error?

Reply:
Already explained in discord the test you did is not how FluxOS works. Opening a port isn't enough, fluxOs creates a service listening on the port where it asks other nodes to check if it's ok from the outside.